### PR TITLE
2879-2921-filter-project-tasks-by-external-system-id

### PR DESCRIPTION
**Pull request by afs25**
Ready for review

### CODE REVIEW

**Comment by afs25** [comment]
Just updated this branch with changes in master @this-andy

File: api/endpoints/progress_process.py [hunk 1]
```diff
@@ -48,37 +48,30 @@ def sort_progress_by_task(progress_dict_):
...
             user_task_assessments = record['count']
             user_task_progress_info_json = json.dumps({'total assessments': user_task_assessments})
-
-
-
-            # updated_user_tasks += pg_utils.execute_non_query(UPDATE_USER_TASK_PROGRESS_SQL, (user_task_progress_info_json, user_id, external_task_id), correlation_id)
-            user_tasks_sql_queries.append((sql_q.UPDATE_USER_TASK_PROGRESS_SQL, (user_task_progress_info_json, user_id, external_task_id)))
-
+            user_tasks_sql_queries.append((sql_q.UPDATE_USER_TASK_PROGRESS_SQL,
+                                           (user_task_progress_info_json, user_id, external_task_id, external_system_name)))
```
Comment (this-andy) [inline api/endpoints/progress_process.py]: Not sure this is the best way to do this.  I'd be tempted to do a query outside the loop to get external_system_id and then pass that to the update query.  This avoids a subquery, replacing it with another clause in the join.  What do you think?

File: api/tests/test_scripts/test_progress.py [hunk 2]
```diff
@@ -33,27 +33,64 @@
 class MyTestCase(test_utils.DbTestCase):
```
Comment (this-andy) [inline api/tests/test_scripts/test_progress.py]: Have you added a test for same external_task_id with different external_system_id?

File: api/tests/test_data/projecttask_data_PSFU.csv [hunk 3]
```diff
@@ -9,4 +9,5 @@ b335c46a-bc1b-4f3d-ad0f-0b8d0826a908,2018-11-05 13:58:07.356280,2018-11-05 13:58
...
-c92c8289-3590-4a85-b699-98bc8171ccde,2018-08-17 12:10:56.986690,2018-08-17 12:10:57.023286,Systematic review for CTG monitoring,active,3ffc498f-8add-4448-b452-4fc7f463aa21,86118f6f-15e9-4c4b-970c-4c9f15c4baf6,2018-08-17 12:00:00.000+00,2018-09-15 12:00:00.000+00,not-open,True,public,,1234,e056e0bf-8d24-487e-a57b-4e812b40c4d8,http://crowd.cochrane.org/index.html,,
\ No newline at end of file
+c92c8289-3590-4a85-b699-98bc8171ccde,2018-08-17 12:10:56.986690,2018-08-17 12:10:57.023286,Systematic review for CTG monitoring,active,3ffc498f-8add-4448-b452-4fc7f463aa21,86118f6f-15e9-4c4b-970c-4c9f15c4baf6,2018-08-17 12:00:00.000+00,2018-09-15 12:00:00.000+00,not-open,True,public,,1234,e056e0bf-8d24-487e-a57b-4e812b40c4d8,http://crowd.cochrane.org/index.html,,
+4ee70544-6797-4e21-8cec-5653c8d5b234,2018-08-17 12:10:57.074321,2018-08-17 12:10:57.111495,Midwife assessment for CTG monitoring,active,3ffc498f-8add-4448-b452-4fc7f463aa21,d92d9935-cb9e-4422-9dbb-65c3423599b1,2018-08-17 09:30:00.000+00,,open,False,public,,ext-6a,9645a6b1-febb-4c40-8ee4-ff3264cf26af,https://www.qualtrics.com,,
```
Comment (afs25) [inline api/tests/test_data/projecttask_data_PSFU.csv]: This row has the same external_task_id as row 8, but a different external_system_id

**Pull request merged**